test/integration/good/compiler-optimizations/cppO1.expected

```diff
@@ -15825,7 +15836,9 @@ class initialize_SoA_model final : public model_base_crtp<initialize_SoA_model>
         Eigen::Matrix<double,-1,1>::Constant(3,
           std::numeric_limits<double>::quiet_NaN());
       out__.write(y);
+      current_statement__ = 2;
       for (int sym1__ = 1; sym1__ <= 4; ++sym1__) {
+        current_statement__ = 2;
```

Comment:
Why is the current statement assignment moved into the for loop? We do not want that

Reply:
I also think this is something the C++ compiler would easily optimize away, but I've manually tried to strip metadata in the correct places to avoid this in the loops we generate for IO